site/app/views/submission/HomeworkView.php

```diff
@@ -1210,14 +1209,198 @@ function downloadFile(file, path) {
+      $threads = $this->core->getQueries()->getRegradeDiscussion($thread_id);
+      $existsStaffPost = false;
+      foreach ($threads as $thread) {
+        if($this->core->getQueries()->isStaffPost($thread['user_id'])) $existsStaffPost = true;
+        break;
+      }
+      $return = <<<HTML
+      <div class = "sub">
+        <div style="float: left; width: 50%"><h3>Regrade Discussion</h3></div>
+HTML;
+        $is_disabled = "";
+        $action = "";
+        $url = "";
+        $class = "btn-default";
+        $deleteMode = false;
+        if($gradeable->getRegradeStatus() === 0){
+          $message = "Request Regrade";
+          $action = "showPopUp()";
+          $deleteMode = false;
+          $url = $this->core->buildUrl(array('component' => 'student',
+                                             'action' => 'request_regrade',
+                                             'gradeable_id' => $gradeable->getId(),
+                                             'student_id' =>$this->core->getUser()->getId()
+                                        ));
+        }else if($gradeable->getRegradeStatus() === -1){
+          if(!$existsStaffPost || $this->core->getUser()->accessGrading()){
+            $message = "Delete Request";
+            $class = "btn-danger";
+            $is_disabled = "";
+            $url = $this->core->buildUrl(array('component' => 'student',
+                                               'action'=> 'delete_request',
+                                               'gradeable_id' => $gradeable->getId(),
+                                               'student_id' => $gradeable->getUser()->getId()
+                                            ));
+            $deleteMode = true;
+          }else{
+            $is_disabled = "disabled";
+            $message = "Request in Review";
+            $url = "";
+            $deleteMode = false;
+          }
+        }else{
+          $message = "Request Reviewed";
+          $is_disabled = "disabled";
+          $url = "";
+          $deleteMode = false;
+        }
+        if(!$deleteMode){
+        $return .= <<<HTML
+          <div style="float:right"><button class = "btn {$class}" onclick="{$action}" {$is_disabled} >$message</button></div>
+HTML;
+        }else{
+          $return .= <<<HTML
+          <div style="float:right">
+            <form method="POST" action="{$url}" id="deleteRequest">
+              <button class = "btn {$class}" type = "submit">$message</button>
+            </form>
+          </div>
+HTML;
+        }
+        $return .= <<<HTML
+        <div class="modal" id="modal-container">
+          <div class="modal-content" id="regradeBox">
+            <h3>Request Regrade</h3>
+            <hr>
+            <p class = "red-message">Warning: Frivoulous requests may result in a grade deduction, loss of late days, or having to retake data structures!</p>
```

Comment:
Any reason this mentions data structures specifically?

Reply:
For the memes

Reply:
👌👀

Reply:
Instructors will have the option to write a specific warning message once the feature to allow/disallow requests is added, it looks half-implemented right now?